digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/NumericalLimitFiller.scala

```diff
@@ -92,23 +92,32 @@ object NumericalLimitFiller {
     * @param changeSet
     * @return
     */
-  def adjustTwoWaySegments(roadLink: RoadLink, assets: Seq[PersistedLinearAsset], changeSet: ChangeSet): (Seq[PersistedLinearAsset], ChangeSet) = {
-    val twoWaySegments = assets.filter(_.sideCode == 1).sortWith(modifiedSort)
-    if (twoWaySegments.length == 1 && assets.forall(_.sideCode == 1)) {
-      val asset = twoWaySegments.head
-      val (adjustedAsset, mValueAdjustments) = adjustAsset(asset, roadLink)
-      (Seq(adjustedAsset), changeSet.copy(adjustedMValues = changeSet.adjustedMValues ++ mValueAdjustments))
-    } else {
-      if (twoWaySegments.length > 1) {
-        val asset = twoWaySegments.head
-        val rest = twoWaySegments.tail
-        val (updatedAsset, newChangeSet) = extend(asset, rest, changeSet)
-        val (adjustedAsset, mValueAdjustments) = adjustAsset(updatedAsset, roadLink)
-        (rest.filterNot(p => newChangeSet.expiredAssetIds.contains(p.id)) ++ Seq(adjustedAsset),
-          newChangeSet.copy(adjustedMValues = newChangeSet.adjustedMValues ++ mValueAdjustments))
-      } else {
-        (assets, changeSet)
-      }
+  def adjustTwoWaySegments(roadLink: RoadLink, sidecode: Int, assets: Seq[PersistedLinearAsset], changeSet: ChangeSet): (Seq[PersistedLinearAsset], ChangeSet) = {
```

Comment:
Maybe we should change this name, because in fact now it adjust any way segment.